File: bin/socval/CoreAutomation.framework/Versions/A/Resources/Scripting/python_automation/coreautomation/xbs_build.py
from CoreAutomationFramework import *

class XBSBuild(object):
    def __init__(self, obj):
        if type(obj) == str:
            self.camBuild = CAMXBSBuild.alloc().initWithString_(obj)
        else:
            self.camBuild = obj
    
    def __str__(self):
        return self.camBuild.description()
        
    def __cmp__(self, other):
        return self.camBuild.compare_(other.camBuild)
    
    @property
    def cam_build(self):
        return self.camBuild
    
    @property
    def build_name(self):
        return self.camBuild.buildName()
    
    @property
    def train_name(self):
        return self.camBuild.train()
    
    @classmethod
    def most_recent_build_for_train(cls,name):
      return CAMXBSBuild.mostRecentBuildForTrain_(name)

File: bin/socval/CoreAutomation.framework/Versions/A/Resources/Scripting/ruby_automation/cam/xbs_build.rb
module CAM
  
  class XBSBuild
    include Comparable
    
    attr_reader :cam_build
    
    def initialize(obj)
      case obj
      when String
        @cam_build = self.class.internal_class_name.alloc.initWithString(obj)
      else
        @cam_build = obj
      end
    end
    
    def build_name
      @cam_build.buildName.to_s()
    end
    
    def train_name
      @cam_build.train.to_s()
    end
    
    def to_s()
      name
    end
    
    def <=>(other)
      @cam_build.compare(other.cam_build)
    end
    
    def name
      @cam_build.description.to_s()
    end
    
    def self.most_recent_build_for_train(name)
      self.new(self.internal_class_name.mostRecentBuildForTrain(name))
    end
    
    protected
    def self.internal_class_name
      if using_macruby?
        CAMXBSBuild
      else
        OSX::CAMXBSBuild
      end
    end
  end
  
end
